models/enterprise_sonic/snmp/sonic_snmp.yml

```
@@ -2,52 +2,281 @@
 GENERATOR_VERSION: '1.0'
 ANSIBLE_METADATA: |
     {
-        'metadata_version': '1.0',
-        'status': ['preview'],
-        'supported_by': 'community',
-        'license': 'Apache 2.0'
+      'metadata_version': '1.1',
+      'status': ['preview'],
+      'supported_by': 'community'
     }
-
 NETWORK_OS: sonic
 RESOURCE: snmp
-COPYRIGHT: Copyright 2020 Dell Inc. or its subsidiaries. All Rights Reserved
+COPYRIGHT: Copyright 2024 Dell Inc. or its subsidiaries. All Rights Reserved.
 DOCUMENTATION: |
-  module: sonic_snmp
-  version_added: 2.10
-  short_description: 'Manages snmp attributes on SONiC.'
-  description: 'This module provides configuration management of SNMP parameters on devices running SONiC'
-  author: "Kumaraguru Narayanan (@nkumaraguru)"
-  notes:
-    - 'Tested against SONiC Version 3.0.1'
-  options:
-    config:
-      description: A dictionary of snmp configurations
-      type: list
-      elements: dict
-      suboptions:
-        community:
-          type: str
-          description: The name or community of the snmp 
-          required: true
-        access:
-          type: str
-          description: The securityName of this community ex.:ro, rw 
-          required: false
-          choices:
-          - ro
-          - rw
-    state:
-      description: The state the configuration should be left in
-      type: str
-      choices:
-      - merged
-      - replaced
-      - overridden
-      - deleted
-      default: merged
+    module: sonic_snmp
+    version_added: 3.3.1
+    short_description: Manage SNMP configuration on SONiC
+    description:
+      - This module provides configuration management of SNMP for devices running SONiC
+    author: Aida Shumburo (@aida-shumburo)
+    options:
+      config:
+        description:
+          - SNMP configuration
+        type: dict
+        suboptions:
+          agentaddress:
+            description:
+              - List of SNMP agent address configuration
```

Constraint between interface and vrf suboptions can be specified here, without repetition for each suboption. 
```suggestion
          agentaddress:
            description:
              - List of SNMP agent address configuration
              - I(interface) and I(vrf) are mutually exclusive.
```